src/azul/indexer/aggregate.py

```diff
@@ -159,10 +161,14 @@ class SetOfDictAccumulator(SetAccumulator):
     """
 
     def accumulate(self, value) -> bool:
-        return super().accumulate(freeze(value))
+        # If the frozen value were a tuple, it would add an extraneous level of
+        # nesting to the accumulated result, because the superclass interprets
+        # tuples as scalars instead of sequences.
```

Comment:
```suggestion
        # tuples as scalars instead of sequences. We use `frozenset` instead. We don't care about retaining duplicates or the ordering of elements.
```

But double-check that this is true. Alternatively, we could use a subclass of tuple. All in all, I am not so sure the sequence_cls thing was a good idea. What do you think?